test/functional/core/startup_spec.lua

```diff
@@ -593,6 +593,36 @@ describe('startup', function()
     eq('  encoding=utf-8\n', fn.system({ nvim_prog, '-n', '-es' }, { 'set encoding', '' }))
   end)
 
+  it('can use -es with a TTY', function()
+    exec([[
+      func Normalize(data) abort
+        " Remove ^M
+        return map(a:data, 'substitute(v:val, "\r", "", "g")')
+      endfunc
+      func OnOutput(id, data, event) dict
+        let g:stdout = Normalize(a:data)
+      endfunc
+      let id = jobstart([v:progpath, '-es'], {
+      \ 'pty': v:true,
+      \ 'stdout_buffered': v:true,
+      \ 'on_stdout': function('OnOutput'),
+      \ })
+      call chansend(id, "put =mode(1)\r")
+      call chansend(id, "%print\r")
+      call chansend(id, "q!\r")
```

Comment:
There may be a data race here. If the output for `%print` is produced before `q!` is received, `cv` may appear before `q!`.

Reply:
I can send the data all at once, which should make it nearly impossible for that to happen since the PTY will echo it all at once as well.

Reply:
I've pushed a change which does that and should also eliminate the possibility of a data race at all.